include/nbl/ui/CWindowWin32.h

```diff
@@ -9,42 +9,44 @@
 namespace nbl {
 namespace ui
 {
-
+class CWindowManagerWin32;
 class CWindowWin32 final : public IWindowWin32
 {
 public:
 	static LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam);
 	static E_KEY_CODE getNablaKeyCodeFromNative(uint8_t nativeWindowsKeyCode);
 
 	//TODO
-	explicit CWindowWin32(core::smart_refctd_ptr<system::ISystem>&& sys, native_handle_t hwnd) : IWindowWin32(std::move(sys))
-	{
-		assert(false);
-		RECT rect;
-		GetWindowRect(hwnd, &rect);
-
-		m_width = rect.right - rect.left;
-		m_height = rect.bottom - rect.top;
-		// TODO m_flags
-	}
+	//explicit CWindowWin32(CWindowManagerWin32* winManager, core::smart_refctd_ptr<system::ISystem>&& sys, native_handle_t hwnd) : IWindowWin32(std::move(sys))
+	//{
+	//	assert(false);
+	//	RECT rect;
+	//	GetWindowRect(hwnd, &rect);
+	//
+	//	m_width = rect.right - rect.left;
+	//	m_height = rect.bottom - rect.top;
+	//	// TODO m_flags
+	//}
 
 	native_handle_t getNativeHandle() const override { return m_native; }
 
-	static core::smart_refctd_ptr<CWindowWin32> create(core::smart_refctd_ptr<system::ISystem>&& sys, int32_t _x, int32_t _y, uint32_t _w, uint32_t _h, E_CREATE_FLAGS _flags)
+	static core::smart_refctd_ptr<CWindowWin32> create(CWindowManagerWin32* winManager, core::smart_refctd_ptr<system::ISystem>&& sys, int32_t _x, int32_t _y, uint32_t _w, uint32_t _h, E_CREATE_FLAGS _flags, const std::string_view& caption)
```

Comment:
why do you need both a pointer to the window manager and the ISystem!?

Reply:
why not?

Reply:
ok, im not sure what ISystem is needed for here

Reply:
> why not?

because I assume the window manager will hold a reference to a IUI and that will hold a pointer to ISystem